src/PerformanceTable.cpp

```diff
@@ -1,49 +1,230 @@
+        for (Perf perf : p) {
+          if (perf.getName() == name) {
+            return perf;
+          }
+        }
+        throw std::invalid_argument("Name not found in performance table");
+      }
     }
+    throw std::invalid_argument("Criterion not found in performance table");
+  } else {
+    throw std::domain_error("Performance table mode corrupted.");
   }
-  throw std::invalid_argument("Row not found in performance table");
 }
 
-std::vector<Performance> PerformanceTable::getPerformanceTable() const {
+std::vector<std::vector<Perf>> PerformanceTable::getPerformanceTable() const {
   return pt_;
-}
\ No newline at end of file
+}
+
+std::string PerformanceTable::getMode() const { return mode_; }
+
+void PerformanceTable::changeMode(std::string mode) {
+  if (mode != "alt" && mode != "crit") {
+    throw std::invalid_argument("Mode must be alt or crit.");
+  }
+  // this operation cannot be done in place, should not be an issue as we should
+  // not call this method often.
+  std::vector<std::vector<Perf>> new_pt;
+  std::map<std::string, int> index;
+  if (mode == "crit") {
+    for (std::vector<Perf> pv : pt_) {
+      for (Perf p : pv) {
+        if (index.count(p.getCrit()) > 0) {
+          // if criterion already seen, add Perf to the right row
+          new_pt[index[p.getCrit()]].push_back(p);
+        } else {
+          // create a new row for this criterion and add Perf
+          std::vector<Perf> v = {p};
+          new_pt.push_back(v);
+          index[p.getCrit()] = new_pt.size() - 1; // index of crit in the new_pt
+        }
+      }
+    }
+  } else {
+    for (std::vector<Perf> pv : pt_) {
+      for (Perf p : pv) {
+        if (index.count(p.getName()) > 0) {
+          // if alternative (or profile) already seen, add Perf to the right row
+          new_pt[index[p.getName()]].push_back(p);
+        } else {
+          // if alternative (or profile) already seen, add Perf to the right row
+          std::vector<Perf> v = {p};
+          new_pt.push_back(v);
+          index[p.getName()] = new_pt.size() - 1; // index of crit in the new_pt
+        }
+      }
+    }
+  }
+
+  pt_ = new_pt;
+  mode_ = mode;
+}
+
+// OPTI could be optimized by using other sort methods if necessary
+void PerformanceTable::sort(std::string mode) {
+  if (mode != "alt" && mode != "crit") {
+    throw std::invalid_argument("Mode must be alt or crit.");
+  }
+  if (mode == "alt" && mode_ != "alt") {
+    this->changeMode("alt");
+  } else if (mode == "crit" && mode_ != "crit") {
+    this->changeMode("crit");
+  }
+
+  for (std::vector<Perf> &pv : pt_) {
+    std::sort(pv.begin(), pv.end(), [](const Perf &a, const Perf &b) {
+      return a.getValue() < b.getValue();
+    });
+  }
+}
+
+std::vector<Perf> PerformanceTable::getAltBetween(std::string crit, float inf,
+                                                  float sup) {
+  if (mode_ != "crit") {
+    throw std::domain_error("Mode must be crit but is currently set to " +
```

Comment:
Lets call the argument std::string with a more meaningful name ie are we talking about criterion, criteria and what ? (ie criterion id, criterion name etc ...)